language/src/lib/y2country/widgets/language_selection.rb

```diff
@@ -0,0 +1,128 @@
+# ------------------------------------------------------------------------------
+# Copyright (c) 2017 SUSE LINUX GmbH, Nuernberg, Germany.
+#
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of version 2 of the GNU General Public License as published by the
+# Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, contact Novell, Inc.
+#
+# To contact Novell about this file by physical or electronic mail, you may find
+# current contact information at www.novell.com.
+# ------------------------------------------------------------------------------
+
+require "yast"
+require "cwm/widget"
+
+Yast.import "Console"
+Yast.import "Language"
+Yast.import "Timezone"
+Yast.import "UI"
+Yast.import "Mode"
+
+module Y2Country
+  module Widgets
+    # Language selection widget
+    class LanguageSelection < CWM::ComboBox
+
+      attr_reader :default
+
+      # @param emit_event [Boolean] flag if handle of widget emit `:redraw` event
+      #   when language changed or not
+      def initialize(emit_event: false)
+        textdomain "country"
+        @default = Yast::Language.language
+        @emit_event = emit_event
+      end
+
+      # Widget label
+      #
+      # @return [String]
+      def label
+        _("&Language")
+      end
+
+      # Widget options
+      #
+      # Widget is forced to report immediatelly after value changed.
+      def opt
+        [:notify, :hstretch]
+      end
+
+      # Initialize the selected value
+      def init
+        self.value = default
+      end
+
+      # Widget help text
+      #
+      # @return [String]
+      def help
+        _(
+          "<p>\n" \
+            "Choose the <b>Language</b> to be used during\n" \
+            "installation and on the installed system.\n" \
+          "</p>\n"
+        )
+      end
+
+      # Handle value changes
+      def handle
+        return if value.nil? || value == default
+        Yast::Timezone.ResetZonemap
+        Yast::Language.Set(value)
+        Yast::Language.languages = Yast::Language.RemoveSuffix(value)
+        @default = value
+        return nil unless @emit_event
+        return nil if Yast::Mode.config
+
+        switch_language
+        :redraw
+      end
+
+      # Store widget value
+      def store
+        handle
+
+        switch_language if !@emit_event && !Yast::Mode.config
```

Comment:
Why not `unless @emit_event || Yast::Mode.Config`? BTW, see my comment above about merging conditions.

Reply:
well, I really dislike unless with more complex condition. because `unless @emit_event || Yast::Mode.Config` is in my head `if !(@emit_event || Yast::Mode.Config)` which I need to translate to `if !@emit_event && !Yast::Mode.Config` so it is three steps computation, so I prefer to be explicit and see it on first sight.
Maybe behavior from languages which does not have unless. So `unless @emit_event` is fine for me as it is simple `if !@emit_event` and is trivial to see, but for more complex expression I have to do computation in my head and it decrease readiness in my head.

Reply:
If you find it more readable, then keep it as it is.